nss-day-cohort-52/rare-client-smoky-pythons
# Implemented subscriber feature to the client

### Issue ticket(s): See [API pull request](https://github.com/nss-day-cohort-52/rare-api-smoky-pythons/pull/67) for tickets

### New file(s): N/A

### Modified File(s): See 28 files in the _files changed_ tab of this PR

### Modifications:
- Got rid of a ton of unnecessary imports and fetch calls
- Added some small styling changes
- Added ternaries to allow us to restrict edit and delete functions to authors only
- Most of the work here is being done on the back end. I just needed to refactor the code to use the new properties we made on our models

### Steps to test:
1. While in the terminal, use git fetch --all in the root directory of this repo
2. Checkout out to `al-subscriptions`
3. Once server and client are running, login to the app.
4. Navigate to the user manager tab and select a user to subscribe to
5. Once subbed to a user, navigate back to the home page by clicking on the main logo in navbar
6. You should now see the posts authored by the user you subbed to.
7. Go back to that user's profile and try to unsubscribe from them.
8. Double check that you are only able to edit and delete things that you are the author of. In other words, you should not be able to edit or delete anybody else's posts or comments
